### PR TITLE
 Fix unhandled errors on maybe_user_id unwraps in authorization logic

### PR DESCRIPTION
**Pull request by mario-nt**
There are several unwraps that lack error handling in the authorization logic when unwrapping the maybe_user_id variable, this PR fix that by handling the errors.

### CODE REVIEW

**Review by josecelano — CHANGES_REQUESTED** [review]
Hi @mario-nt, in all these cases, the feature requires an authenticated user. It would not be possible to allow guest users to perform those actions because it's hardcoded in the feature. 

For example, the image proxy works with a user quota, therefore it would not work without a user.

Instead of panicking I would return an "unauthenticated" error, even before calling the authorization service.

And maybe we should even check for invalid permissions in the casbin policy. For example, these policies should not be allowed:

```
guest, GetImageByUrl
guest, AddTorrent
guest, ChangePassword
```

In order to allow them in the future the app behaviour must be changed. For the ChangePassword action it does not make sense at all to include it in the policies unless we want to allow in the the future the "admin" to change other users' passwords, which is something I wouldn't do.

File: src/services/proxy.rs [hunk 1]
```diff
@@ -38,14 +38,16 @@ impl Service {
     /// * The image URL is not an image.
     /// * The image is too big.
     /// * The user quota is met.
-    #[allow(clippy::missing_panics_doc)]
     pub async fn get_image_by_url(&self, url: &str, maybe_user_id: Option<UserId>) -> Result<Bytes, Error> {
+        let Some(user_id) = maybe_user_id else {
+            return Err(Error::Unauthenticated);
```
Comment (josecelano) [inline src/services/proxy.rs:43]: Hi @mario-nt, should not this also be `ServiceError::UnauthorizedActionForGuests`?

**Review by josecelano — CHANGES_REQUESTED** [review]

File: src/services/proxy.rs (hunk 1 above)
Comment (mario-nt) [reply src/services/proxy.rs:43]: Hi @josecelano, in this case, I didn't want to change the function's signature and also, that error it is already implemented and returns the text "Sign in to see image":

https://github.com/torrust/torrust-index/blob/720b2fa537676fc2b34c3c0b986599c12e695a61/src/web/api/server/v1/contexts/proxy/handlers.rs#L30-L39

https://github.com/torrust/torrust-index/blob/720b2fa537676fc2b34c3c0b986599c12e695a61/src/ui/proxy.rs#L19-L46

**Review by josecelano — APPROVED** [review]

**Comment by josecelano** [comment]
ACK 2e4b8d61d5b972eedae4615f70958560d43a810e

**Pull request merged**